editor/src/components/canvas/controls/new-canvas-controls.tsx

```diff
@@ -269,15 +309,173 @@ const NewCanvasControlsInner = (props: NewCanvasControlsInnerProps) => {
     'NewCanvasControlsInner',
   )
 
-  const { localSelectedViews, localHighlightedViews, setLocalSelectedViews } = props
+  const {
+    localSelectedViews,
+    localHighlightedViews,
+    setLocalSelectedViews,
+    setLocalHighlightedViews,
+    setSelectionAreaRectangle,
+  } = props
   const cmdKeyPressed = keysPressed['cmd'] ?? false
 
   const contextMenuEnabled = !isLiveMode(editorMode)
   const { maybeHighlightOnHover, maybeClearHighlightsOnHoverEnd } = useMaybeHighlightElement()
 
-  const { onMouseMove, onMouseDown, onMouseUp } = useSelectAndHover(
-    cmdKeyPressed,
-    setLocalSelectedViews,
+  const selectModeHooks = useSelectAndHover(cmdKeyPressed, setLocalSelectedViews)
+
+  const ref = React.useRef<HTMLDivElement | null>(null)
+
+  const [mousePoint, setMousePoint] = React.useState<CanvasPoint | null>(null)
```

Comment:
It seems like you've confused `WindowPoint` with `CanvasPoint` when dealing with the `mousePoint`, since it is being stored as a `CanvasPoint` but is actually a `WindowPoint`. But I'd say we don't need to (or want to) track the `mousePoint` in a `useState` here. We only need to use it inside `onMouseDown` (where we have access to the data on the event) and when updating the `selectionArea` (which should probably be happening inside `onMouseMove` rather than as a `useMemo`).

Reply:
>  confused `WindowPoint` with `CanvasPoint`

Is that a root cause of the various hard-to-repro bugs I experienced last week?

Reply:
my bad, indeed I got confused